### ISSUE TITLE
Wishlist: Include some kind of cycle detection facilities for obo graphs (obo files)

### ISSUE THREAD

**Issue by matentzn** [INCATools/ontology-development-kit#255]

**Comment by cmungall** [comment]
for subclass cycles, just use sparql?

**Comment by matentzn** [comment]
Yeah I was looking more for something like a verification mechanism that an ontology "is acyclic", which appears to be a requirement for the basic artefact.. sparql is probably the way to go.

**Comment by cmungall** [comment]
OK, so this is more general that just subclass cycles (which should be flagged automatically, as we will infer equivalence between named classes). You need to query over the basic graph.

Or what about this: if an ontology is basic, then create a root property P that is a superproperty of all object properties in the ontology. Declare P to be irreflexive, transitive. Then query for any C SubClassOf P some D & D SubClassOf P some C. May need EMR
